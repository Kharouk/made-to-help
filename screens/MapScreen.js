import React from 'react';
import MapView from 'react-native-maps';
import { View } from 'react-native';
import CurrentLocation from '../components/CurrentLocation';
import DynamicLocation from '../components/DynamicLocation';

export default class MapScreen extends React.Component {
  static navigationOptions = {
    title: 'Map',
  };

  render() {
    return (
      <View style={{ flex: 1 }} testID="map">
          <DynamicLocation />
      </View>
    );
  }
}
