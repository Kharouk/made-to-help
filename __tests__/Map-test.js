import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MapScreen from '../screens/MapScreen';

// Mocks the react-native-maps element which has our maps:
jest.mock('react-native-maps', () => {
  const React = require('react');
  return class MockMapView extends React.Component {
    static Marker = props => React.createElement('Marker', props, props.children);

    render() {
      return React.createElement('MapView', this.props, this.props.children);
    }
  };
});


  describe('Map snapshot', () => {
    let navigation;
    beforeEach(() => {
      navigation = jest.fn();
    })
    it('renders successfully', async () => {
      // const tree = renderer.create(<MapScreen/>).toJSON();
      // expect(tree).toMatchSnapshot();

  });
});
