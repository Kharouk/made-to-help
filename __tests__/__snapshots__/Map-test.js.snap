// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map snapshot renders successfully 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
  testID="map"
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <MapView
      provider="google"
      showsUserLocation={true}
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <Callout>
      <View
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.3)",
            "width": 500,
          }
        }
      >
        <Text
          accessibilityLabel="Hello! Journey Time is 0 Minutes and the distance is 
 null KM"
          style={
            Object {
              "color": "hotpink",
              "fontSize": 40,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          0 Minutes 
 null KM
        </Text>
      </View>
    </Callout>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 10,
          "position": "absolute",
          "right": 10,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
        />
      </View>
    </Callout>
  </View>
</View>
`;
